core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -927,7 +935,7 @@ public void handleResponse(TransportResponse.Empty vResponse) {
                     public void handleException(TransportException exp) {
                         onReplicaFailure(nodeId, exp);
                         logger.trace("[{}] transport failure during replica request [{}], action [{}]", exp, node, replicaRequest, transportReplicaAction);
-                        if (ignoreReplicaException(exp) == false) {
+                        if (mustFailReplica(exp)) {
```

Comment:
can we assert that ignoreReplicaException is false? it will be weird to have something fail a shard but not reported.